dashboard/lib/policies/child_account.rb

```diff
@@ -85,27 +85,34 @@ def self.pre_lockout_user?(user)
     user.created_at < lockout_date
   end
 
-  # The date on which the student's account will be locked if the account is not compliant.
-  def self.lockout_date(user)
-    return if compliant?(user)
-
+  # The date on which the student's grace period ends.
+  # @param user [User] the student account
+  # @param approximate [Boolean] if true, return an approximate date if the exact date is not known
+  def self.grace_period_end_date(user, approximate: false)
     user_state_policy = state_policy(user)
     return unless user_state_policy
 
-    # CAP non-compliant students who were created:
-    # - before the policy took effect - should be locked out during the all users lockout phase.
-    # - after the policy took effect - should be locked out immediately.
-    user_predates_policy?(user) ? user_state_policy[:lockout_date] : user_state_policy[:start_date]
+    grace_period_duration = user_state_policy[:grace_period_duration]
+    return unless grace_period_duration
+
+    start_date = DateTime.parse(user.child_account_compliance_state_last_updated) if ComplianceState.grace_period?(user)
+    start_date = user_state_policy[:lockout_date] if approximate && start_date.nil?
```

Comment:
```suggestion
    start_date ||= user_state_policy[:lockout_date] if approximate
```
This would be the same, right?

Reply:
Correct. I wasn't sure which variant to use, so I chose the one I found easier to understand from the first reading.

And here I used another approach:
https://github.com/code-dot-org/code-dot-org/blob/359e66f2be0c0bf68cf4da7383b2048516f96b26/dashboard/app/helpers/child_account_helper.rb#L9

Reply:
Cool, I just saw us using `||=` a lot, so I felt like I should point out we could do it here too. Just a nit-pic.